### ISSUE TITLE
multiple SendTrig UGens crash supernova

### ISSUE THREAD

**Issue by dyfer** [supercollider/supercollider#3339]
Hello,
I came across a surprising crash with supernova and I traced it down to running multiple SendTrig UGens in a Parallel Group. I'm on the recent develop build on macOS 10.12.
The following will crash on my machine after few triggers of SendTrig (I assume this indeterminacy is because the synths are started asynchronously). I would really appreciate if others confirmed this happening on different systems as well. Reproducer:

```
Server.supernova;
s.boot;

(
fork{
	~mainGroup = ParGroup.new;
	// ~mainGroup = Group.new; //doesn't crash in a regular group
	s.sync;
	~def = OSCdef(\update, {|msg| postf("id: %, time: %\n", msg[2], msg[3].round(0.01))}, '/tr'); //post triggers to see progress
	~trigSynths = 4.collect({|doInc| //crash happens for me with 3 and more SendTrig synths
		{SendTrig.kr(Impulse.kr(4), doInc, Sweep.kr)}.play(target: ~mainGroup); //the faster the impulse, the sooner the crash happens
	});
}
)

~trigSynths.do(_.free); ~def.free;
```

**Comment by jamshark70** [comment]
Confirmed -- it crashes very quickly for me in Ubuntu Studio 16.04, 3.9 beta branch.

I'll take a guess that the SendTrig UGen code assumes it could never possibly be constructing multiple outgoing messages at the same time.

However, SendReply is unaffected -- I'm running the test code now with the following substitution:

```
		{SendReply.kr(Impulse.kr(4), '/tr', doInc, Sweep.kr)}.play(target: ~mainGroup);
```

... and it's up to `id: 50` and counting, no crash.

Of course SendTrig should be fixed, but there is a workaround: use SendReply.

(TBH, SendReply is more flexible -- I don't understand why anybody is still using SendTrig. IMO SendTrig may be a candidate for deprecation.)

**Comment by jamshark70** [comment]
Hm... it would be easy to deprecate the SendTrig UGen:

```
SendTrig : UGen {
	*ar { arg in = 0.0, id = 0, value = 0.0;
		^SendReply.ar(in, '/tr', value, id);
	}
	*kr { arg in = 0.0, id = 0, value = 0.0;
		^SendReply.kr(in, '/tr', value, id);
	}
	...
}
```

Removing SendTrig from plugins would break precompiled SynthDefs and probably other clients, but this would avoid the present crash.

**Comment by dyfer** [comment]
Thanks, indeed SendReply doesn't have this problem!

I'd be in favor of deprecating SendTrig, especially that it's easy to create backwards compatibility workaround.

**Comment by dyfer** [comment]
Fixed in #4379
